TenureInformationApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs

```diff
@@ -220,5 +226,307 @@ private async Task InsertDatatoDynamoDB(TenureInformation entity)
             await _dynamoDb.SaveAsync(entity.ToDatabase()).ConfigureAwait(false);
             _cleanup.Add(async () => await _dynamoDb.DeleteAsync(entity.ToDatabase()).ConfigureAwait(false));
         }
+
+        [Fact]
+        public async Task EditTenureDetailsWhenTenureDoesntExistReturnsNull()
+        {
+            var mockQuery = _fixture.Create<TenureQueryRequest>();
+            var mockRequestObject = _fixture.Create<EditTenureDetailsRequestObject>();
+            var mockRawBody = "";
+
+            // call gateway method
+            var response = await _classUnderTest.EditTenureDetails(mockQuery, mockRequestObject, mockRawBody).ConfigureAwait(false);
+
+            // assert result = null
+            response.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task EditTenureDetailsWhenNoChangesAreInRequestObjectDatabaseIsntUpdated()
+        {
+            // create mock tenure
+            var mockTenure = _fixture.Build<TenureInformation>()
+                .With(x => x.StartOfTenureDate, DateTime.UtcNow)
+                .With(x => x.EndOfTenureDate, DateTime.UtcNow)
+                .Create();
+
+            // insert mock tenure into database
+            await InsertDatatoDynamoDB(mockTenure).ConfigureAwait(false);
+
+            var mockQuery = _fixture.Build<TenureQueryRequest>().With(x => x.Id, mockTenure.Id).Create(); // with tenure id
+            var mockRequestObject = new EditTenureDetailsRequestObject();
+            var mockRawBody = "";
+
+            var updaterResponse = new UpdateEntityResult<TenureInformationDb>(); // with no changes
+
+            // setup updater
+            _mockUpdater
+                .Setup(x => x.UpdateEntity(It.IsAny<TenureInformationDb>(), It.IsAny<string>(), It.IsAny<EditTenureDetailsRequestObject>()))
+                .Returns(updaterResponse);
+
+            // call gateway method
+            var response = await _classUnderTest.EditTenureDetails(mockQuery, mockRequestObject, mockRawBody).ConfigureAwait(false);
+
+            // assert response is UpdateEntityResult
+            response.Should().BeOfType(typeof(UpdateEntityResult<TenureInformationDb>));
+
+            // load entity from database
+            var databaseResponse = await _dynamoDb.LoadAsync<TenureInformationDb>(mockTenure.Id).ConfigureAwait(false);
+
+            // assert entity wasnt updated (matches inserted data)
+            databaseResponse.StartOfTenureDate.Should().BeCloseTo((DateTime) mockTenure.StartOfTenureDate, 1.Seconds());
```

Comment:
I did not know there were `int` extension methods for specifying time periods - every day is a school day.

Reply:
https://fluentassertions.com/datetimespans/